libs/deal-ticket/src/components/deal-ticket/deal-ticket.tsx

```diff
@@ -63,6 +65,14 @@ export const DealTicket = ({
   const order = watch();
   // When order state changes persist it in local storage
   useEffect(() => setPersistedOrder(order), [order, setPersistedOrder]);
+
+  // When switch back to market order, set price to market price
+  useEffect(() => {
```

Comment:
chore (blocking): can you show me why we need this? The order estimate query now gets the correct price directly from market.data.markPrice or market.data.indicativePrice (or order.price if limit order).

This also results in loss of state of your user input limit order price see below:

Uploading Nov-24-2022 16-13-07.mp4…

Reply:
I see where the issue is. Will push a fix